custom_components/waste_collection_schedule/waste_collection_schedule/source/mid_devon_gov_uk.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import time
+from datetime import datetime
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Mid Devon District Council"
+DESCRIPTION = (
+    "Source for waste collection services for Mid Devon District Council"
+)
+URL = "https://www.middevon.gov.uk"
+
+TEST_CASES = {
+    "Cullompton": {"uprn": 100040354099},
+    "Cullompton - string": {"uprn": "100040354099"},  # Knutsford is in Cheshire East and shouldn't have any collection information!
+}
+
+ICON_MAP = {
+    "Domestic": "mdi:trash-can",
+    "Food": "mdi:food-apple",
+    "Recycling": "mdi:recycle",
+    "Garden": "mdi:leaf",
+}
+
+API_URLS = {
+    "session": "https://my.middevon.gov.uk/en/AchieveForms/?form_uri=sandbox-publish://AF-Process-2289dd06-9a12-4202-ba09-857fe756f6bd/AF-Stage-eb382015-001c-415d-beda-84f796dbb167/definition.json&redirectlink=%2Fen&cancelRedirectLink=%2Fen&consentMessage=yes",
+    "auth": "https://my.middevon.gov.uk/authapi/isauthenticated?uri=https%253A%252F%252Fmy.middevon.gov.uk%252Fen%252FAchieveForms%252F%253Fform_uri%253Dsandbox-publish%253A%252F%252FAF-Process-927a4f8b-67a7-4c41-8e39-00479c300a63%252FAF-Stage-eb382015-001c-415d-beda-84f796dbb167%252Fdefinition.json%2526redirectlink%253D%25252Fen%2526cancelRedirectLink%253D%25252Fen%2526consentMessage%253Dyes&hostname=my.middevon.gov.uk&withCredentials=true",
+    "authResp": "https://my.middevon.gov.uk/apibroker/runLookup?id=645e14020c9cc&repeat_against=&noRetry=false&getOnlyTokens=undefined&log_id=&app_name=AchieveForms",
+    "serviceTypes": "https://my.middevon.gov.uk/apibroker/runLookup?id=645e14020c9cc&repeat_against=&noRetry=false&getOnlyTokens=undefined&log_id=&app_name=AchieveForms",
+    "schedule": "https://my.middevon.gov.uk/apibroker/runLookup?id=645e14020c9cc&repeat_against=&noRetry=false&getOnlyTokens=undefined&log_id=&app_name=AchieveForms",
+}
+
+HEADERS = {
+    "user-agent": "Mozilla/5.0",
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Source:
+    def __init__(self, uprn: str):
+        self._uprn = str(uprn)
+
+    def fetch(self):
+        s = requests.Session()
+
+        # Get session cookies
+        r0 = s.get(API_URLS["session"], headers=HEADERS)
+        r0.raise_for_status()
+
+        # Get session key from the PHPSESSID (in the cookies)
+        authRequest = s.get(API_URLS["auth"], headers=HEADERS)
+        authData = authRequest.json()
+        sessionKey = authData["auth-session"]
+
+        now = time.time_ns() // 1_000_000
+        urlNonce = str(now)
+
+        # Get AuthenticateResponse nonce
+        authRespRequest = s.post(
+            API_URLS["authResp"] + "&_" + urlNonce + "&sid=" + sessionKey,
+            headers=HEADERS,
+        )
+        authenticateResponseNonce = self.achieveFormsData(authRespRequest)["0"][
```

Comment:
Based on your comment with tests failing to run, I guess the` ["0"]` should be `[0]`